metadata-service/auth-impl/src/main/java/com/datahub/authentication/authenticator/DataHubTokenAuthenticator.java

```diff
@@ -23,27 +26,37 @@
  *
  * This authenticator requires the following configurations:
  *
- *  - signingAlgorithm (optional): the algorithm used to verify JWT's. This should be THE SAME ONE used by the {@link TokenService}. Defaults to HS256.
+ *  - signingAlgorithm (optional): the algorithm used to verify JWT's. This should be THE SAME ONE used by the {@link StatelessTokenService}. Defaults to HS256.
  *  - signingKey: a key used to sign all JWT tokens using the provided signingAlgorithm
  */
 @Slf4j
 public class DataHubTokenAuthenticator implements Authenticator {
 
   static final String SIGNING_KEY_CONFIG_NAME = "signingKey";
+  static final String SALTING_KEY_CONFIG_NAME = "saltingKey";
   static final String SIGNING_ALG_CONFIG_NAME = "signingAlg";
   static final String DEFAULT_SIGNING_ALG = "HS256";
   static final String DEFAULT_ISSUER = "datahub-metadata-service";
 
   // Package-Visible for testing.
-  TokenService tokenService;
+  StatefulTokenService _statefulTokenService;
 
   @Override
-  public void init(@Nonnull final Map<String, Object> config) {
+  public void init(@Nonnull final Map<String, Object> config, @Nonnull final Map<String, Object> context) {
     Objects.requireNonNull(config, "Config parameter cannot be null");
     final String signingKey = Objects.requireNonNull((String) config.get(SIGNING_KEY_CONFIG_NAME), "signingKey is a required config");
+    final String saltingKey = Objects.requireNonNull((String) config.get(SALTING_KEY_CONFIG_NAME), "saltingKey is a required config");
```

Comment:
One other option is to place the salt into the AccessTokenMetadata itself in an encrypted fashion.. This won't prevent the replay attack though

Reply:
Let's address offline